process/smartContract/process.go

```diff
@@ -444,14 +457,58 @@ func (sc *scProcessor) executeSmartContractCall(
 	return vmOutput, nil
 }
 
+func (sc *scProcessor) isSCROnlyInformational(scr data.TransactionHandler) bool {
+	if scr.GetValue().Cmp(zero) > 0 {
+		return false
+	}
+
+	function, _, err := sc.argsParser.ParseCallData(string(scr.GetData()))
+	if err != nil {
+		return true
+	}
+
+	if core.IsSmartContractAddress(scr.GetRcvAddr()) {
+		return false
+	}
+
+	_, err = sc.builtInFunctions.Get(function)
+	if err != nil {
+		return true
+	}
+
+	return false
+}
+
+func (sc *scProcessor) cleanSCRDataWithOnlyInfo(scrs []data.TransactionHandler) ([]data.TransactionHandler, []*vmcommon.LogEntry) {
+	cleanedUPSCrs := make([]data.TransactionHandler, 0)
+	logsFromSCRs := make([]*vmcommon.LogEntry, 0)
+
+	for _, scr := range scrs {
+		if sc.isSCROnlyInformational(scr) {
+			logsFromSCRs = append(logsFromSCRs, createNewLogFromSCR(scr))
+			continue
+		}
+
+		cleanedUPSCrs = append(cleanedUPSCrs, scr)
+	}
+
+	if sc.flagCleanUpSCRData.IsSet() {
```

Comment:
I think the condition isn't ok. we should return cleaned SCRs if the flag is set

Reply:
done